OtripleS.Web.Api.Tests.Unit/Services/StudentSemesterCourseServiceTests/StudentSemesterCourseServiceTests.Exceptions.Retrieve.cs

```diff
@@ -0,0 +1,102 @@
+//---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE AS LONG AS SOFTWARE FUNDS ARE DONATED TO THE POOR
+//----------------------------------------------------------------
+
+using System;
+using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore;
+using Moq;
+using OtripleS.Web.Api.Models.SemesterCourses.Exceptions;
+using OtripleS.Web.Api.Models.StudentSemesterCourses.Exceptions;
+using Xunit;
+
+namespace OtripleS.Web.Api.Tests.Unit.Services.StudentSemesterCourseServiceTests
+{
+    public partial class StudentSemesterCourseServiceTests
+    {
+        [Fact]
+        public void ShouldThrowDependencyExceptionOnRetrieveAllWhenSqlExceptionOccursAndLogIt()
+        {
+            // given
+            SqlException sqlException = GetSqlException();
+
+            var expectedStudentSemesterCourseDependencyException =
+                new StudentSemesterCourseDependencyException(sqlException);
+
+            this.storageBrokerMock.Setup(broker => broker.SelectAllStudentSemesterCourses())
+                .Throws(sqlException);
+
+            //when. then
+            Assert.Throws<StudentSemesterCourseDependencyException>(() =>
+                this.studentSemesterCourseService.RetrieveAllStudentSemesterCourses());
+
+            this.loggingBrokerMock.Verify(broker =>
+                    broker.LogCritical(It.Is(SameExceptionAs(expectedStudentSemesterCourseDependencyException))),
+                        Times.Once);
+
+            this.storageBrokerMock.Verify(broker => broker.SelectAllStudentSemesterCourses(),
+                Times.Once);
+
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public void ShouldThrowDependencyExceptionOnRetrieveAllWhenDbExceptionOccursAndLogIt()
+        {
+            // given
+            var databaseUpdateException = new DbUpdateException();
+
+            var expectedSemesterCourseDependencyException =
+                new SemesterCourseDependencyException(databaseUpdateException);
```

Comment:
Shouldnt that be a StudentSemesterCourseDependencyException?

Reply:
Yes, Sorry, I had so many typing mistakes regarding this two entities. I will fix this